File: _site/technical/leet/detectCapitalUse.py
# problem 520
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if all(x.isupper() for x in word) or all(x.islower() for x in word) or (all(x.islower() for x in word[1:]) and word[0].isupper()):
            return True
        return False

File: technical/leet/detectCapitalUse.cpp
// problem 520
#include "string"
using namespace std;

class Solution {
public:
    bool detectCapitalUse(string word) {
        int capCount = 0;
        for(char& c: word){
            if(isupper(c)){
                capCount++;
            }
        }

        // all upper or lower
        if(capCount == word.length() || capCount == 0){
            return(true);
        }
        // camel case
        else if(isupper(word[0]) && capCount == 1){
            return(true);
        }
        else {
            return(false);
        }
    }
};